File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        def memoizeStairs(memo, n):
            if n in memo:
                return memo[n]
            else:
                memo[n] = memoizeStairs(memo, n - 1) + memoizeStairs(memo, n - 2)
                return memo[n]
        memo = {}
        memo[1] = 1
        memo[2] = 2
        return memoizeStairs(memo, n)

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */

var stairsHelper = function(n, memo) {
    if (n in memo) {
        return memo[n];
    }
    memo[n] = stairsHelper(n - 1, memo) + stairsHelper(n - 2, memo);
    return memo[n];
}

var climbStairs = function(n) {
    let memo = {};
    memo[0] = 0;
    memo[1] = 1;
    memo[2] = 2;
    return stairsHelper(n, memo);
};